tree.go

```diff
@@ -618,7 +618,7 @@ func (n *InternalNode) Delete(key []byte, resolver NodeResolverFn) (bool, error)
 			// as well.
 			for _, c := range n.children {
 				if _, ok := c.(Empty); !ok {
-					break
+					return false, nil
```

Comment:
While breaking was reasonable, the right thing is to return `false`. Only `break` would mean L625 being executed signaling that a node deletion should happen, when that shouldn't be the case.